### PR TITLE
feat: make module discovery works without entry-points

### PR DESCRIPTION
**Pull request by oux**
Signed-off-by: Sebastien MICHEL <sebastien.t.michel@renault.com>

### CODE REVIEW

File: repo/subcmds/__init__.py [hunk 1]
```diff
@@ -14,15 +14,100 @@
...
+else:
+  for modname in iter_namespace(__name__):
+    name = modname.rpartition('.')[2]
+    clsn = name.capitalize()
+    while clsn.find('_') > 0:
+      h = clsn.index('_')
+      clsn = clsn[0:h] + clsn[h + 1:].capitalize()
+
+    import importlib
+    mod = importlib.import_module(modname)
```
Comment (tardyp) [inline repo/subcmds/__init__.py:92]: the whole point of iter_entry_points is to not load the modules (and their dependencies).

Then you are building here a heuristic to map classnames to subcommand names which looks like very fragile.

I doubt I will be able to upstream this when the time comes.

**Review by tardyp — CHANGES_REQUESTED** [review]

File: repo/subcmds/__init__.py (hunk 1 above)
Comment (oux) [reply repo/subcmds/__init__.py:92]: 1/ I only added a fallback when iter_entry_points() has no return
2/ This fallback code is based on [original code](https://github.com/tardyp/google-repo/commit/ea11d8c6d44960b6b01455d089bfe52b3b423509#diff-5db94cc59acc7877994e1dcf547b9899155e6e1f536f8c91b7ccf22c823eb29aL33), would you prefere I use `__import__()` ?

Comment (tardyp) [reply repo/subcmds/__init__.py:92]: In this case, this indeed makes sense. I think this deserve a comment.

File: repo/subcmds/__init__.py [hunk 2]
```diff
@@ -14,15 +14,100 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import os
+import sys
 import pkg_resources
+import pkgutil
+import sys
+
+from typing import Generator
```
Comment (tardyp) [inline repo/subcmds/__init__.py]: This would be the only typed code in repo so I think we can remove that

File: repo/subcmds/__init__.py [hunk 3]
```diff
@@ -14,15 +14,100 @@
...
+
+  Yields the name of all modules found at a given full-qualified module name.
+  Use 'importlib.import_module' to import the found submodules.
+
+  Usage:
+
+  .. code-block:: python
+
+      import importlib
+      from alton_utils.pyinstaller import iter_namespace
```
Comment (tardyp) [inline repo/subcmds/__init__.py]:
```suggestion
```

File: repo/subcmds/__init__.py [hunk 4]
```diff
@@ -14,15 +14,65 @@
...
 # limitations under the License.
 
-import os
+import sys
 import pkg_resources
+import pkgutil
+import sys
+
+def _get_module_by_name(namespace: str):
+  return sys.modules[namespace]
```
Comment (tardyp) [inline repo/subcmds/__init__.py]: what is the point of this function? I think direct access is fine

**Review by tardyp — CHANGES_REQUESTED** [review]

File: setup.py [hunk 5]
```diff
@@ -20,7 +20,7 @@
 
 setup(
   name="gitrepo",
-  version="2.8.0",
+  version="2.9.0",
```
Comment (tardyp) [inline setup.py]: versionning matches upstream. dont bump like that

**Review by tardyp — APPROVED** [review]

**Pull request merged**